src/exercises-final/02.extra.js

```diff
@@ -0,0 +1,44 @@
+// Compound Components
+
+import React from 'react'
+import {Switch} from '../switch'
+
+class Toggle extends React.Component {
+  static On = ({on, children}) => (on ? children : null)
+  static Off = ({on, children}) => (on ? null : children)
+  static Button = ({on, toggle, ...props}) => (
+    <Switch on={on} onClick={toggle} {...props} />
+  )
+  state = {on: false}
+  toggle = () =>
+    this.setState(
+      ({on}) => ({on: !on}),
+      () => this.props.onToggle(this.state.on),
+    )
+  render() {
+    return React.Children.map(this.props.children, child => {
+      if (typeof child.type === 'function')
```

Comment:
Could we improve this further by checking that `child.type` is one of `Toggle.On`, `Toggle.Off`, or `Toggle.Button`?

Reply:
Alright I did. I was trying to figure out how to do that before, but I couldn't. I think I have it now.